processor/probabilisticsamplerprocessor/logsprocessor_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package probabilisticsamplerprocessor
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+func TestNewLogsProcessor(t *testing.T) {
+	tests := []struct {
+		name         string
+		nextConsumer consumer.Logs
+		cfg          *Config
+		wantErr      bool
+	}{
+		{
+			name: "nil_nextConsumer",
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(component.NewID(typeStr)),
+				SamplingPercentage: 15.5,
+			},
+			wantErr: true,
+		},
+		{
+			name:         "happy_path",
+			nextConsumer: consumertest.NewNop(),
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(component.NewID(typeStr)),
+				SamplingPercentage: 15.5,
+			},
+		},
+		{
+			name:         "happy_path_hash_seed",
+			nextConsumer: consumertest.NewNop(),
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(component.NewID(typeStr)),
+				SamplingPercentage: 13.33,
+				HashSeed:           4321,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newLogsProcessor(context.Background(), componenttest.NewNopProcessorCreateSettings(), tt.nextConsumer, tt.cfg)
+			if tt.wantErr {
+				assert.Nil(t, got)
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func TestLogsSampling(t *testing.T) {
+	tests := []struct {
+		name     string
+		cfg      *Config
+		received int
+	}{
+		// {
```

Comment:
Will that commented-out code eventually be uncommented? It might be better to remove them now and add them later, once they are needed.